File: model/src/main/java/net/artux/pda/model/StatusModel.java
package net.artux.pda.model;

import net.artux.pda.model.quest.story.StoryDataModel;

import java.io.Serializable;

import lombok.Data;

@Data
public class StatusModel implements Serializable {

    private boolean success;
    private String description;
    private StoryDataModel storyDataModel;

    public StatusModel() {
    }

    public StatusModel(String message) {
        success = true;
        description = message;
    }

    public StatusModel(Throwable throwable) {
        success = false;
        description = throwable.getMessage();
    }
}

File: model/src/main/java/net/artux/pda/model/StatusModel.kt
package net.artux.pda.model

import net.artux.pda.model.quest.story.StoryDataModel
import java.io.Serializable

class StatusModel : Serializable {
    var success = false
    var description: String? = null
    var storyDataModel: StoryDataModel? = null

    constructor() {}
    constructor(message: String?) {
        success = true
        description = message
    }

    constructor(throwable: Throwable) {
        success = false
        description = throwable.message
    }
}